src/Features/CSharp/Portable/QuickInfo/CSharpSemanticQuickInfoProvider.cs

```diff
@@ -88,6 +88,17 @@ protected override ImmutableArray<TaggedText> TryGetNullabilityAnalysis(Workspac
 
             var typeInfo = semanticModel.GetTypeInfo(bindableParent, cancellationToken);
 
+            if (typeInfo.Type.IsValueType)
+            {
+                return default;
+            }
+
+            switch (symbolInfo.Symbol)
+            {
+                case IFieldSymbol { HasConstantValue: true }: return default;
+                case ILocalSymbol { HasConstantValue: true }: return default;
```

Comment:
nifty!

Reply:
@alrz want to right a feature that looks for things like `case IField field when field.HasConstantValue: ` and convert to this form?   Or cases of patterns like `is Whatever w && <additional checks on w>`?

Reply:
Actually that form was what @ryzngard was trying to write but we couldn't remember then "when" syntax... :-)

Reply:
haha.  i like this better!

Reply:
>case IField field when field.HasConstantValue:

https://github.com/dotnet/roslyn/pull/32386